src/Language/PureScript/TypeChecker/Monad.hs

```diff
@@ -13,23 +13,43 @@ import Control.Monad.Error.Class (MonadError(..))
 import Control.Monad.State
 import Control.Monad.Writer.Class (MonadWriter(..), censor)
 
+import Data.List (intercalate)
 import Data.Maybe
 import qualified Data.Map as M
-import Data.Text (Text)
+import Data.Text (Text, isPrefixOf, unpack)
 import qualified Data.List.NonEmpty as NEL
 
 import Language.PureScript.Crash (internalError)
 import Language.PureScript.Environment
 import Language.PureScript.Errors
-import Language.PureScript.Kinds
 import Language.PureScript.Names
+import Language.PureScript.Pretty.Types
+import Language.PureScript.Pretty.Values
 import Language.PureScript.TypeClassDictionaries
 import Language.PureScript.Types
-
--- | A substitution of unification variables for types or kinds
+import Text.PrettyPrint.Boxes (render)
+
+newtype UnkLevel = UnkLevel (NEL.NonEmpty Unknown)
+  deriving (Eq, Show)
+
+-- This instance differs from the NEL instance in that longer but otherwise
```

Comment:
I think you could express this as `deriving Ord via (Down (NEL.NonEmpty Unknown))`

Reply:
No, I don't think so. I don't want to reverse the ordering in general. The ordering is
```
[1, 2, 3]
[1, 2]
[1]
[2, 3, 4]
[2, 3]
[2, 4]
[2]
```